### PR TITLE
bump: upgrade golang version to v1.23

### PR DESCRIPTION
**Pull request by Two-Hearts**

### CODE REVIEW

**Review by priteshbandi — DISMISSED** [review]
LGTM

**Review by vaninrao10 — APPROVED** [review]
LGTM

File: test/e2e/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/notaryproject/notation/test/e2e
 
-go 1.21
+go 1.22
```
Comment (shizhMSFT) [inline test/e2e/go.mod:3]: I'm curious why the golang version for e2e test is not the same as the main one?

**Review by shizhMSFT — COMMENTED** [review]
The dependency bump up should also be cherry-picked or backported to `v1.2.0`.

**Review by shizhMSFT — CHANGES_REQUESTED** [review]
It seems #538 did the right thing and this PR does not.

File: test/e2e/go.mod (hunk 1 above)
Comment (Two-Hearts) [reply test/e2e/go.mod]: Updated.

**Comment by JeyJeyGao** [comment]
what about `building.md`?

> Building above binaries require [golang](https://golang.org/dl/) with version `>= 1.22`.

**Comment by Two-Hearts** [comment]
> what about `building.md`?
> 
> > Building above binaries require [golang](https://golang.org/dl/) with version `>= 1.22`.

This should be correct? `>= 1.22` includes 1.23.

**Comment by shizhMSFT** [comment]
> what about `building.md`?
> 
> > Building above binaries require [golang](https://golang.org/dl/) with version `>= 1.22`.

Good catch. Since `go.mod` requires `1.23`, we should update the `building.md` for `1.23`. This is also for security purposes.

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**